multiwallet.go

```diff
@@ -56,80 +52,62 @@ func NewMultiWallet(rootDir, dbDriver, netType, politeiaHost string) (*MultiWall
-	}
-
-	// init database for saving/reading wallet objects
-	err = mwDB.Init(&Wallet{})
-	if err != nil {
-		log.Errorf("Error initializing wallets database: %s", err.Error())
-		return nil, err
-	}
-
-	// init database for saving/reading proposal objects
-	err = mwDB.Init(&Proposal{})
+	dcrDB, dcrRootDir, err := initializeDCRWallet(rootDir, dbDriver, netType)
 	if err != nil {
-		log.Errorf("Error initializing wallets database: %s", err.Error())
-		return nil, err
+		log.Errorf("error initializing DCRWallet: %s", err.Error())
+		return nil, errors.Errorf("error initializing DCRWallet: %s", err.Error())
 	}
 
 	mw := &MultiWallet{
-		dbDriver:    dbDriver,
-		rootDir:     rootDir,
-		db:          mwDB,
-		chainParams: chainParams,
-		wallets:     make(map[int]*Wallet),
-		badWallets:  make(map[int]*Wallet),
-		syncData: &syncData{
-			syncProgressListeners: make(map[string]SyncProgressListener),
+		DbDriver:    dbDriver,
+		RootDir:     dcrRootDir,
+		DB:          dcrDB,
+		ChainParams: chainParams,
+		Assets: &Assets{
+			DCR: struct {
+				Wallets     map[int]*dcr.Wallet
+				BadWallets  map[int]*dcr.Wallet
+				DBDriver    string
+				RootDir     string
+				DB          *storm.DB
+				ChainParams *chaincfg.Params
+			}{
+				Wallets:     make(map[int]*dcr.Wallet),
+				BadWallets:  make(map[int]*dcr.Wallet),
+				DBDriver:    dbDriver,
+				RootDir:     dcrRootDir,
+				DB:          dcrDB,
+				ChainParams: chainParams,
+			},
 		},
-		txAndBlockNotificationListeners:  make(map[string]TxAndBlockNotificationListener),
-		accountMixerNotificationListener: make(map[string]AccountMixerNotificationListener),
-	}
-
-	mw.Politeia, err = newPoliteia(mw, politeiaHost)
-	if err != nil {
-		return nil, err
 	}
 
 	// read saved wallets info from db and initialize wallets
-	query := mw.db.Select(q.True()).OrderBy("ID")
-	var wallets []*Wallet
+	query := mw.DB.Select(q.True()).OrderBy("ID")
+	var wallets []*dcr.Wallet
 	err = query.Find(&wallets)
 	if err != nil && err != storm.ErrNotFound {
 		return nil, err
 	}
 
 	// prepare the wallets loaded from db for use
 	for _, wallet := range wallets {
-		err = wallet.prepare(rootDir, chainParams, mw.walletConfigSetFn(wallet.ID), mw.walletConfigReadFn(wallet.ID))
-		if err == nil && !WalletExistsAt(wallet.dataDir) {
+		err = wallet.Prepare(mw.RootDir, mw.ChainParams, mw.walletConfigSetFn(wallet.ID), mw.walletConfigReadFn(wallet.ID))
+		if err == nil && !WalletExistsAt(wallet.DataDir) {
 			err = fmt.Errorf("missing wallet database file")
 		}
 		if err != nil {
-			mw.badWallets[wallet.ID] = wallet
+			mw.Assets.DCR.BadWallets[wallet.ID] = wallet
 			log.Warnf("Ignored wallet load error for wallet %d (%s)", wallet.ID, wallet.Name)
 		} else {
-			mw.wallets[wallet.ID] = wallet
+			mw.Assets.DCR.Wallets[wallet.ID] = wallet
 		}
+
+		logLevel := wallet.ReadStringConfigValueForKey(LogLevelConfigKey, "")
+		SetLogLevels(logLevel)
+
+		// initialize Politeia.
+		wallet.NewPoliteia(politeiaHost)
```

Comment:
wallet.NewPoliteia returns some values that currently ignored.
Also, politeia shouldn't be a wallet construct - it's not tied to any one wallet. It should be much like the `internal/vsp` package that is pretty much standalone but can work with any provided dcr wallet.

Reply:
that true, also i agree we shouldn't have politeia as a wallet struct, current problem is that politeia is now a wallet property, viewable when a dcr wallet is opened, and each dcr wallet can have it's own specific settings in regards to politeia.

but with the new internal politeia package that was merged, we'd have to take away this code anyways